glommio/src/sys/source.rs

```diff
@@ -323,6 +323,7 @@ impl Source {
 impl Drop for Source {
     fn drop(&mut self) {
         let mut inner = self.inner.borrow_mut();
+        inner.wakers.waiters.clear();
```

Comment:
that's the kind of things bugs are made of. "No one is interested" until they are, and dropping the source was an accident.

How often did you find pending sources at this point ? I'd be interested in understanding why they happen in the first place.

Reply:
Btw I don't feel strongly about this and I'd be happy to drop that commit if you're concerned about it. My point, though, is that if this is causing trouble, then it would be indicative of a (probably worse) underlying issue.

Reply:
I agree that this is indicative of a bigger problem, the main issue is how things manifest.
Things silently not executing is certainly not fun to debug - as you know.

My suggested strategy with this would be:
1. Print the existing wakers, if any, first.
2. Figure out why those wakers exist and try, if possible, to patch the callers not to have them.
3. Assert that there are no more wakers at drop, at least in debug mode.